fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -5601,6 +5603,150 @@ private CompletableFuture<Void> preloadMetaData() {
                 return AsyncUtil.DONE;
             }
         }
+
+        /**
+         * In the event that a store has been corrupted, and the header has been lost, this method can be used to open
+         * the store, and fill in the missing header.
+         * <p>
+         * If it is possible that multiple different instances have a different idea of any of the versions
+         * ({@link #setFormatVersion format version}, {@link RecordMetaData#getVersion() metaData version}, or
+         * {@link #getUserVersion() user version}), ensure that this builder is created with the maximal option for each.
```

Comment:
```suggestion
         * If it is possible that multiple different instances have a different idea of any of the versions
         * ({@linkplain #setFormatVersion format version}, {@linkplain RecordMetaData#getVersion() metaData version}, or
         * {@linkplain #getUserVersion() user version}), ensure that this builder is created with the maximal option for each.
```

I'm also not totally sure I get what it's getting at. Is it trying to say that the `FDBRecordStore.Builder` needs to have the maximum possible meta-data, format, and user version in order to ensure that we don't go accidentally make any of those go backwards? I think that's sensible, and there's probably not a way around that, at some level

Reply:
Yeah, that is what it is getting at. I'm not sure how to word it differently, but I can try....